packages/venia-ui/lib/components/Routes/routes.js

```diff
@@ -1,15 +1,19 @@
 import React, { Suspense, lazy } from 'react';
-import { Route, Switch } from 'react-router-dom';
+import { Route, Switch, useLocation } from 'react-router-dom';
 
 import { fullPageLoadingIndicator } from '../LoadingIndicator';
 import MagentoRoute from '../MagentoRoute';
+import { useScrollTopOnChange } from '@magento/peregrine/lib/hooks/useScrollTopOnChange';
 
 const CartPage = lazy(() => import('../CartPage'));
 const CheckoutPage = lazy(() => import('../CheckoutPage'));
 const CreateAccountPage = lazy(() => import('../CreateAccountPage'));
 const Search = lazy(() => import('../../RootComponents/Search'));
 
 const Routes = () => {
+    const { pathname } = useLocation();
+    useScrollTopOnChange(pathname);
```

Comment:
Some other examples I saw out on the web included also observing `search`. I'm not sure if we want to implement that here or in the components that actually care about search such as pagination or the search/filter components.

Reply:
I think it makes sense to do it here. Those components are changing `search` but scrolling is a page-level concern; I don't think they should know / care / control the scrolling.

Reply:
I think it makes sense to do it here. Those components are changing `search` but scrolling is a page-level concern; I don't think they should know / care / control the scrolling.

Reply:
Agree with @supernova-at here. I wouldn't scroll to top when search params change.

Reply:
> We'd have to accept that it'll happen in every case where we ever change search...and I don't have that level of confidence. In contrast, I do have that level of confidence when it comes to pathname.

To be specific, the `Router` component can confidently control scrolling on path changes but not search changes. That said, other components such as the Category or Search pages should be able to utilize the new hook to handle scrolling to top as they do already.

```js
// CURRENT
const Category = props => {
    const { search } = useLocation();
    const filtersFromSearch = getFilters(search);
    useEffect(() => {
        runFilterQuery()
        window.scrollTop();
    }, [filtersFromSearch])    
}
// PROPOSED
const Category = props => {
    const { search } = useLocation();
    const filtersFromSearch = getFilters(search);
    useScrollTopOnChange(filtersFromSearch);
    useEffect(() => {
        runFilterQuery()
    }, [filtersFromSearch])
}
```